lib/src/merged_tree.rs

```diff
@@ -959,7 +959,14 @@ impl MergedTreeBuilder {
                         }
                     }
                 }
-                Ok(MergedTreeId::Legacy(tree_builder.write_tree()))
+                let legacy_id = tree_builder.write_tree();
+                if store.use_tree_conflict_format() {
+                    let legacy_tree = store.get_tree(&RepoPath::root(), &legacy_id)?;
+                    let merged_tree = MergedTree::from_legacy_tree(legacy_tree)?;
+                    Ok(merged_tree.id())
+                } else {
+                    Ok(MergedTreeId::Legacy(legacy_id))
+                }
```

Comment:
nit: API doc has to be updated since legacy tree may be upgraded

Reply:
Fixed.